### ISSUE TITLE
 Members are not shown even when there is a functionality of 'All members' in 'who can post'

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19884]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to dev web chrome
2. Create a new workspace and invite few members
3. Go to search and click on the workspace chat with square logo
4. Click on header and there is only 1 in the members
5. But if clicked on settings and clicked on 'who can post' , it displays ' All members' option also.
6. Thus, if 'All members' option is shown, the newly invited users should have been added on the 'members' section

## Expected Result:
Members count should be shown since there is a functionality of 'All members' in 'who can post'

## Actual Result:
Members are not shown even when there is a functionality of 'All members' in 'who can post'

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.20.1

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/c4be93a7-fb05-4f6c-a22c-7d8a442c1d5d

https://github.com/Expensify/App/assets/93399543/d6a98e0d-e551-4c46-b709-eef7f027ef3a

**Expensify/Expensify Issue URL:**

**Issue reported by:** avi-shek-jha

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685179945958949

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by JmillsExpensify** [comment]
Just a bit of feedback, I found the reproduction steps very hard to follow. In fact, I am still having trouble understand the issue that you're referring to here.

> Members count should be shown since there is a functionality of 'All members' in 'who can post'

Are you referring to the members that show on the members page within the settings for that specific chat report? If so, I am going to put this issue on hold, as we will begin refactoring this page very shortly.

**Comment by avi-shek-jha** [comment]
Hi. What I wanted to say is. When I have added let's say 3 members on the workspace (using invite members), then It should have shown the members count as 3 on the new workspace chat page also.(To cross verify, after you click on header and go to settings, we can also see the option on 'Who can post' as ''All members". Thus, since 'All members' functionality is there, then I assumed that the new workspace chat should have mentioned all the users of the workspace on the members section)

**Comment by JmillsExpensify** [comment]
Can you please suggest a more specific set of reproduction steps using the titles of the pages/right-hand pane in the app? This is what you currently have. 

> Go to dev web chrome
> Create a new workspace and invite few members
> Go to search and click on the workspace chat with square logo
> Click on header and there is only 1 in the members
> But if clicked on settings and clicked on 'who can post' , it displays ' All members' option also.
> Thus, if 'All members' option is shown, the newly invited users should have been added on the 'members' section

**Comment by avi-shek-jha** [comment]
1. Go to dev web chrome and login with user A
2. Click on Settings > Workspace > Create a New Workspace (let's say Workspace test) > Invite members > Invite User B and User C. Thus, now this newly created workspace has 3 members in total. (User A, User B and User C )
3. Now, go to LHN and search for the workspace chat (that has a square logo) we created in step 2. 
4. Make sure you're on the workspace chat page.
5. Click on header of the workspace chat so that 'Details ' on right hand side pane appears
6. Notice that in 'Members' , it is shown as count 1.
7. Click on members and notice that it's still just showing the admin (User A) in the workspace. We had 3 members in total (User A, User B and User C) since we invited User B and User C to the workspace as referenced in step 2. Thus, the members count should be 3 in this case.

https://github.com/Expensify/App/assets/123831727/952cbf6e-128e-49a4-9c9e-332028b584fa

**Comment by JmillsExpensify** [comment]
Thank you very much for the reproduction steps. I don't think this is bug and we should close this issue.

Here's the reason: A workspace chat is between each member on the workspace, and all admins on that workspace. So if a workspace has 10 members and 2 admins. The there will be 12 workspace chats (admins also get their own workspace chat).

That last one is the case you are seeing. You are the only admin on the workspace. So your workspace chat will be you with yourself. For any members, their workspace chat will have two members, them and then you as the admin.  

Here's that case in my test account. I'm the only admin and this is my workspace chat.
![Screenshot 2023-06-08 at 22 24 54](https://github.com/Expensify/App/assets/1127863/77cce4a4-6331-4ac1-bee7-01f1a86993b0)

Then here's the workspace chat for the member. ![Screenshot 2023-06-08 at 22 25 41](https://github.com/Expensify/App/assets/1127863/2b631fa6-5333-47bc-823f-b3719a3e51db). 

Everything is by design in both cases.

**Comment by JmillsExpensify** [comment]
I'm going to close as a result, though please let me know if you have any follow-on questions.
